### PR TITLE
THF-170: Site search

### PR DESCRIPTION
**Pull request by xkhaven**
- New site wide search for test environment
- Translations missing (THF-425)
- Search suggestions not implemented

### CODE REVIEW

File: pages/search.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+              )
+            })}
+
+          { pages > 1 && 
+            <Pagination
+              language={locale as any}
+              onChange={(event, index) => {
+                event.preventDefault()
+                setPageIndex(index)
+                setTimeout(() => {
```
Comment (Jussiles) [inline pages/search.tsx:213]: Is timeout needed here?

File: pages/search.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      ...(await serverSideTranslations(locale, ['common'])),
+    },
+    revalidate: REVALIDATE_TIME
+  }
+}
+
+const ItemBlock = (props: any): JSX.Element => {
+  const content = props.result
+  const title = content.title[0]
+  const text = content.field_lead_in === undefined
```
Comment (Jussiles) [inline pages/search.tsx:57]: It's not so easy to understand how the text variable value is get

File: pages/api/search.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  if (req.method !== 'GET') {
+    res.status(400)
+    return  
+  }
+
+  const { q, index, locale }:QueryParams = req?.query as any
+  const elastic = Elastic.getElasticClient()
+  const size = 20
+  const from: number = index > 0 ? size * index : 0
+  const body: any = {
```
Comment (Jussiles) [inline pages/api/search.ts:26]: Body could be easily typed

**Review by Jussiles — APPROVED** [review]
Works well! 🦜 added couple of small comments to consider

File: pages/search.tsx (hunk 1 above)
Comment (xkhaven) [reply pages/search.tsx:213]: On some cases scrolling to up when pager changes didn't work. Assuming the list wasn't loaded yet so added little timeout to fix this.

File: pages/search.tsx (hunk 2 above)
Comment (xkhaven) [reply pages/search.tsx]: Updated code to more readable.

File: pages/api/search.ts (hunk 3 above)
Comment (xkhaven) [reply pages/api/search.ts]: Typed as `MsearchMultisearchBody`.

**Pull request merged**